### PR TITLE
Ensure that input in recently open files modal is focused

### PR DESCRIPTION
**Pull request by krassowski**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/16194

## Code changes

- [x] adds a new expectation in the test to not rely on subtle styling changes on snapshots
- [x] fixes the issue by traversing into the shadow DOM

## User-facing changes

The search box in the recently closed modal gets properly focused after opening it.

## Backwards-incompatible changes

None

### CODE REVIEW

File: galata/test/documentation/general.test.ts-snapshots/running-modal-documentation-linux.png
Comment (krassowski) [inline galata/test/documentation/general.test.ts-snapshots/running-modal-documentation-linux.png:1]: The focus indicator is not pretty but this is out of scope for this PR.

**Review by fcollonval — APPROVED** [review]
Thanks @krassowski
